### PR TITLE
feat: make analysis buttons open workflow landing pages (#137)

### PR DESCRIPTION
**Pull request by hunterckx**
- ~~No error handling for the API request~~
- ~~Only regulation is set up~~

### CODE REVIEW

File: app/utils/galaxy-api.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import ky from "ky";
+
+interface WorkflowLandingsBody {
+  request_state: { reference_genome: string };
```
Comment (mvdbeek) [inline app/utils/galaxy-api.ts:4]: This is going to vary across workflows, I suppose
```suggestion
  request_state: { [key: string]: string }:
```
would be the right interface ?

File: app/utils/galaxy-api.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+export async function getWorkflowLandingUrl(
+  workflowId: string,
+  referenceGenome: string
+): Promise<string> {
+  const body: WorkflowLandingsBody = {
+    request_state: {
+      reference_genome: referenceGenome,
+    },
+    workflow_id: workflowId,
+    workflow_target_type: "trs_url",
```
Comment (mvdbeek) [inline app/utils/galaxy-api.ts:34]:
```suggestion
    workflow_target_type: "trs_url",
    public: true,
```

File: app/utils/galaxy-api.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  };
+  const res = await ky.post<WorkflowLanding>(WORKFLOW_LANDINGS_API_URL, {
+    json: body,
+    retry: {
+      methods: ["post"],
+    },
+  });
+  const id = (await res.json()).uuid;
+  return WORKFLOW_LANDING_URL_PREFIX + encodeURIComponent(id);
```
Comment (mvdbeek) [inline app/utils/galaxy-api.ts:43]:
```suggestion
  return `${WORKFLOW_LANDING_URL_PREFIX}${id}?public=true;
```

**Review by mvdbeek — COMMENTED** [review]
Pretty cool thank you!

**Comment by hunterckx** [comment]
@mvdbeek Thanks for the feedback! I've:
- Updated the types to make more room for alternative keys (though in a different way than suggested -- I wanted to make it explicit what the keys can be)
- Added `public: true` to the API request body
- Added `?public=true` to the constructed landing page URL (but left the URL encoding in -- although it's not necessary in theory, I'm inclined to be cautious and default to encoding things when constructing URLs)

**Comment by NoopDog** [comment]
@hunterckx, can you rebase this, please? Thanks!

File: app/apis/catalog/brc-analytics-catalog/common/entities.ts [hunk 4]
```diff
@@ -25,3 +34,8 @@ export interface EntitiesResponsePagination {
   size: number;
   total: number;
 }
+
+export enum WORKFLOW_ID {
+  REGULATION = "https://dockstore.org/api/ga4gh/trs/v2/tools/#workflow/github.com/iwc-workflows/ChIPseq_PE/main",
```
Comment (mvdbeek) [inline app/apis/catalog/brc-analytics-catalog/common/entities.ts:39]:
```suggestion
  REGULATION = "https://dockstore.org/api/ga4gh/trs/v2/tools/#workflow/github.com/iwc-workflows/chipseq-pe/main/versions/v0.12",
```

File: app/apis/catalog/brc-analytics-catalog/common/entities.ts [hunk 5]
```diff
@@ -25,3 +34,8 @@ export interface EntitiesResponsePagination {
...
+  TRANSCRIPTOMICS = "https://dockstore.org/api/ga4gh/trs/v2/tools/#workflow/github.com/iwc-workflows/rnaseq-pe/main",
```
Comment (mvdbeek) [inline app/apis/catalog/brc-analytics-catalog/common/entities.ts:40]:
```suggestion
  TRANSCRIPTOMICS = "https://dockstore.org/api/ga4gh/trs/v2/tools/#workflow/github.com/iwc-workflows/rnaseq-pe/main/versions/v0.9",
```

**Review by mvdbeek — COMMENTED** [review]
We're expecting the full TRS url here, which is the TRS id and the version.

**Comment by mvdbeek** [comment]
For right now this is working well. I have a few things to build on here, could we get this merged ?

**Pull request merged**